src/notifications/use-push-notifications.tsx

```diff
@@ -53,21 +53,20 @@ export const NotificationContextProvider: React.FC = ({children}) => {
 
   const register = useCallback(
     async (enabled: boolean) => {
+      if (!fcmToken) return;
       try {
-        const token = await messaging().getToken();
-        if (!token) return;
-        setFcmToken(token);
         mutateRegister({
-          token,
+          token: fcmToken,
           language: getLanguageAndTextEnum(language),
           enabled,
         });
-        return token;
+        return fcmToken;
       } catch (e) {
         Bugsnag.notify(`Failed to register for push notifications: ${e}`);
       }
     },
-    [language, mutateRegister],
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [language, fcmToken],
```

Comment:
adding `mutateRegister` here was causing fire this register function multiple times, is not needed, so the solution was to add this ESLint exception rule. fcmToken became part of the dependencies.